src/teleop_twist_joy.cpp

```diff
@@ -153,6 +153,114 @@ TeleopTwistJoy::TeleopTwistJoy(const rclcpp::NodeOptions& options) : Node("teleo
   }
 
   pimpl_->sent_disable_msg = false;
+
+  auto param_callback = 
+  [this](std::vector<rclcpp::Parameter> parameters)
+  {
+    auto result = rcl_interfaces::msg::SetParametersResult();
+    result.successful = true;
+
+    // Loop to check if changed parameters are of expected data type
+    for(const auto & parameter : parameters)
+    {
+      /// If parameter is axis_linear, axis_angular, enable_button or enable_turbo_button
+      if (
+        (parameter.get_name().find("axis") != std::string::npos) ||
+        (parameter.get_name().find("enable") != std::string::npos)
+      )
+      {
+        if (parameter.get_type() != rclcpp::ParameterType::PARAMETER_INTEGER)
+        {
+          RCLCPP_ERROR(this->get_logger(),"Only integer values can be set for \"%s\". "
+          "You entered type: %s. Rejecting change...",
+          parameter.get_name().c_str(),
+          parameter.get_type_name().c_str());
+          result.successful = false;
+          return result;
+        }        
+      }
+      else
+      {
+        if (parameter.get_type() != rclcpp::ParameterType::PARAMETER_DOUBLE)
+        {
+          RCLCPP_ERROR(this->get_logger(),"Only double values can be set for \"%s\". "
+          "You entered type: %s. Rejecting change...",
+          parameter.get_name().c_str(),
+          parameter.get_type_name().c_str());
+          result.successful = false;
+          return result;
+        }
+      }
+    }
+
+    // If result.successful is true
+    std::string prefix;
+
+    // Loop to assign changed parameters to the member variables
+    for (const auto & parameter : parameters)
+    {        
+      // If parameter is axis_linear.x, axis_linear.y or axis_linear.z 
+      if (parameter.get_name().find("axis_linear") != std::string::npos)
```

Comment:
While I get that you are trying to reduce the amount of duplication with the `find` here, I think it makes the code much harder to reason about.  It's also slower since you are doing more string manipulation with the `substr` stuff below.  Instead I'll suggest that you just have a separate `if` statement for each of the parameters.  It's more duplication, but it will also be easier to make sure it is correct.

Reply:
Okay, will do